File: app/src/main/java/com/example/unima_l003/demoyoutubeplayer/views/YoutubeFrameLayout.java
package com.example.unima_l003.demoyoutubeplayer.views;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.FrameLayout;

public class YoutubeFrameLayout extends FrameLayout {

    public YoutubeFrameLayout(Context context) {
        super(context);
    }

    public YoutubeFrameLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public YoutubeFrameLayout(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
        int width = getMeasuredWidth();
        int height = width / 16 * 9;
        setMeasuredDimension(width, height);

    }
}

File: app/src/main/java/com/example/unima_l003/demoyoutubeplayer/views/YoutubeFrameLayout.kt
package com.example.unima_l003.demoyoutubeplayer.views

import android.content.Context
import android.util.AttributeSet
import android.widget.FrameLayout

class YoutubeFrameLayout : FrameLayout {

    constructor(context: Context) : super(context) {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
        val width = measuredWidth
        val height = width / 16 * 9
        setMeasuredDimension(width, height)

    }
}
